src/main/java/ru/sbt/mipt/oop/event_handlers/DoorEventHandler.java

```diff
@@ -0,0 +1,41 @@
+package ru.sbt.mipt.oop.event_handlers;
+
+import ru.sbt.mipt.oop.home.Room;
+import ru.sbt.mipt.oop.home.SmartHome;
+import ru.sbt.mipt.oop.door.Door;
+
+import static ru.sbt.mipt.oop.event_handlers.SensorEventType.DOOR_OPEN;
+
+public class DoorEventHandler {
+    private final SensorEvent event;
+    private final SmartHome smartHome;
+
+    public DoorEventHandler(SensorEvent event, SmartHome smartHome) {
+        this.event = event;
+        this.smartHome = smartHome;
+    }
+
+    public SmartHome handleDoorEvent() {
+
+        for (Room room : smartHome.getRooms()) {
+            for (Door door : room.getDoors()) {
+                if (door.getId().equals(event.getObjectId())) {
+                    if (event.getType() == DOOR_OPEN) {
+                        System.out.println("Door " + door.getId() + " in room " + room.getName() + " was opened.");
+                        door.setOpen(true);
+                    } else {
+                        if (room.getName().equals("hall")){
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D1%80%D0%B0%D0%B7%D0%B4%D0%B5%D0%BB%D1%8F%D0%B9%D1%82%D0%B5-%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BE%D0%B1%D1%8A%D0%B5%D0%BA%D1%82%D0%BE%D0%B2-%D0%B8-%D1%81%D1%86%D0%B5%D0%BD%D0%B0%D1%80%D0%B8%D0%B8-%D1%83%D0%BC%D0%BD%D0%BE%D0%B3%D0%BE-%D0%B4%D0%BE%D0%BC%D0%B0